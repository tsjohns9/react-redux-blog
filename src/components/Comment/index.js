import React from 'react';

const Comment = ({ name, body }) => {
  const shortName = new Set(name.split(' ')).values();
  const first = shortName.next().value;
  const last = shortName.next().value;

  const full = last
    ? first.charAt(0).toUpperCase() +
      first.slice(1) +
      ' ' +
      last.charAt(0).toUpperCase() +
      last.slice(1)
    : first.charAt(0).toUpperCase() + first.slice(1);

  return (
    <div style={{ borderBottom: '1px solid #eee' }}>
      <p className="my-4">
        <b>{full}</b>
      </p>
      <p>{body}</p>
    </div>
  );
};

export default Comment;
