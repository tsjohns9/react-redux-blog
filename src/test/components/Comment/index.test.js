import React from 'react';
import { shallow } from 'enzyme';
import Comment from '../../../components/Comment';

test('should render Header correctly', () => {
  console.log(Comment);
  const wrapper = shallow(
    <Comment name={'trevor johnson'} body={'adsf fdsa kmckd ak;iw'} />
  );
  console.log(wrapper);
  expect(wrapper).toMatchSnapshot();
});
