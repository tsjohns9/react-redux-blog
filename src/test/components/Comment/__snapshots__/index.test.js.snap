// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div
  style={
    Object {
      "borderBottom": "1px solid #eee",
    }
  }
>
  <p
    className="my-4"
  >
    <b>
      Trevor Johnson
    </b>
  </p>
  <p>
    adsf fdsa kmckd ak;iw
  </p>
</div>
`;
